app/src/main/java/it/quip/android/model/Notification.java

```diff
@@ -7,79 +9,213 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import com.parse.FindCallback;
 import com.parse.ParseException;
 import com.parse.ParseObject;
 import com.parse.ParseClassName;
 import com.parse.ParsePush;
+import com.parse.ParseQuery;
 import com.parse.SendCallback;
 
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import it.quip.android.listener.NotificationHandler;
+import it.quip.android.listener.ParseModelQueryHandler;
+import it.quip.android.util.TimeUtils;
+
 
 @ParseClassName("Notification")
 public class Notification extends ParseObject implements Parcelable {
 
     public static final int STANDARD_NOTIFICATION = 0;
-    public static final String PUSH_TEXT_BODY_KEY = "alert";
-    public static final String PUSH_SENDER_ID = "sender";
-    public static final String PUSH_CIRCLE_KEY = "circle";
+    public static final String PUSH_TEXT_BODY_KEY = "text_body";
+    public static final String PUSH_SENDER_ID = "sender_uid";
+    public static final String PUSH_IMAGE_URL_KEY = "image_url";
+    public static final String PUSH_RECEIVER_ID = "receiver_uid";
     public static final String PUSH_TYPE_KEY = "notification_type";
-
-    private long uid;
-    private String text;
-    private String notificationImageUrl;
-    private int timestamp;
-    private int type;  // TODO: define enum
-    private boolean viewed;
-
-    public boolean isViewed() {
-        return viewed;
+    public static final String PUSH_TIMESTAMP_KEY = "timestamp";
+    public static final String PUSH_VIEWED_KEY = "viewed";
+    public static final String PUSH_CHANNEL_KEY = "channel";
+
+    private String mSenderUid;
+    private String mReceiverUid;
+    private String mText;
+    private String mNotificationImageUrl;
+    private String mChannel;
+
+    private long mTimestamp;
+    private int mType;  // TODO: define enum
+    private boolean mViewed;
+
+    public String getChannel() {
+        mChannel = this.getString(PUSH_CHANNEL_KEY);
+        return mChannel;
     }
 
-    public void setViewed(boolean viewed) {
-        this.viewed = viewed;
+    public void setChannel(String channel) {
+        this.safePutKey(PUSH_CHANNEL_KEY, channel);
+        this.mChannel = channel;
     }
 
-    public long getUid() {
-        return uid;
+    public boolean viewed() {
```

Comment:
viewed is non java its very pythonic, but reads so much better.  still will change